crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -546,6 +555,15 @@ def upgrade_compute_nodes(virt)
           return
         end
 
+        # remove upgraded compute nodes
+        compute_nodes.select! { |n| !n.upgraded? }
```

Comment:
same comment as with `non_founder_nodes`. I'd prefer you assign the new set to a new variable, like `non_upgraded_compute_nodes`.